libraries/dagster-iceberg/kitchen_sink.py

```diff
@@ -0,0 +1,57 @@
+import polars as pl
+from dagster import (
+    AllPartitionMapping,
+    AssetIn,
+    AssetSpec,
+    Definitions,
+    StaticPartitionsDefinition,
+    asset,
+)
+from dagster_iceberg.config import IcebergCatalogConfig
+from dagster_iceberg.io_manager.polars import IcebergPolarsIOManager
+from dagster_polars import PolarsParquetIOManager
+
+
+NUM_PARTS = 16  # 0
+
+CATALOG_NAME = "default"
+WAREHOUSE_PATH = "/tmp/warehouse"
+NAMESPACE = "default"
+
+parts = StaticPartitionsDefinition([f"part.{i}" for i in range(NUM_PARTS)])
+raw_nyc_taxi_data = AssetSpec(
+    key="nyc.parquet", partitions_def=parts
+).with_io_manager_key("polars_parquet_io_manager")
+
+
+@asset(
+    ins={
+        "raw_nyc_taxi_data": AssetIn(
+            "nyc.parquet", partition_mapping=AllPartitionMapping()
+        )
+    },
+    io_manager_key="iceberg_polars_io_manager",
+)
+def clean_nyc_taxi_data(raw_nyc_taxi_data: dict[str, pl.LazyFrame]) -> pl.LazyFrame:
+    return pl.concat(raw_nyc_taxi_data.values())
+
+
+catalog_config = IcebergCatalogConfig(
+    properties={
+        "uri": f"sqlite:///{WAREHOUSE_PATH}/iceberg_catalog.db",
+        "warehouse": f"file://{WAREHOUSE_PATH}",
+    }
+)
+
+
+defs = Definitions(
+    assets=[raw_nyc_taxi_data, clean_nyc_taxi_data],
+    resources={
+        "polars_parquet_io_manager": PolarsParquetIOManager(
+            base_dir="https://storage.googleapis.com/anaconda-public-data/nyc-taxi/"
```

Comment:
We actually self-host some taxi parquet files, but it might not work with `base_dir` like being done here.

https://github.com/dagster-io/project-dagster-university/blob/dc644d2a0e469cf91612a9a2b433304c810e24b9/course/pages/dagster-essentials/lesson-3/asset-materialization.md?plain=1#L19

The `anaconda-public-data` is probably pretty stable, but it might not hurt to consider a self-hosted alternative at some point.

Reply:
> We actually self-host some taxi parquet files, but it might not work with `base_dir` like being done here.
> 
> https://github.com/dagster-io/project-dagster-university/blob/dc644d2a0e469cf91612a9a2b433304c810e24b9/course/pages/dagster-essentials/lesson-3/asset-materialization.md?plain=1#L19

These are actually not self-hosted, but the ones coming directly from NYC.gov!

Initially, I wanted to load from GCS (not using `https://` URLs), but I'm blocked by https://github.com/pola-rs/polars/issues/21344. I think could consider moving to those in the future, but this works for now and has been around for a long time.

Reply:
Thanks though; that actually reminded me to remove the `gcsfs` dependency!